Add io.github.neph_io.dsost

Neph-IO opened the pull request:
<!-- ⚠️⚠️ Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Go to the preview tab to click the links below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

- [ x] Please describe the application briefly. <!-- This PR adds the Flatpak manifest, icon, metainfo, and related files for the DSOST player, a Dark Souls OST player. -->
- [ x] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [ x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [x ] I have [built][build] and tested the submission locally.
- [ x] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:**

<!-- ⚠️⚠️ DO NOT modify anything below this line ⚠️⚠️  -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

bbhtt commented on com.github.neph_io.dsost.yml:
```diff
@@ -0,0 +1,39 @@
+      - install -Dm644 ./flatpak_build/com.github.neph_io.dsost.desktop /app/share/applications/com.github.neph_io.dsost.desktop
+      - install -Dm644 ./flatpak_build/com.github.neph_io.dsost.png /app/share/icons/hicolor/256x256/apps/com.github.neph_io.dsost.png
+      - install -Dm644 com.github.neph_io.dsost.metainfo.xml /app/share/metainfo/com.github.neph_io.dsost.metainfo.xml
+    sources:
+      - type: dir
+        path: ..
+
```
how will the ci here fetch sources from your PC?

bbhtt commented on com.github.neph_io.dsost.yml:
```diff
@@ -0,0 +1,39 @@
+      - install -Dm755 -t /app/data ./data/*
+      - install -Dm644 ./flatpak_build/com.github.neph_io.dsost.desktop /app/share/applications/com.github.neph_io.dsost.desktop
+      - install -Dm644 ./flatpak_build/com.github.neph_io.dsost.png /app/share/icons/hicolor/256x256/apps/com.github.neph_io.dsost.png
+      - install -Dm644 com.github.neph_io.dsost.metainfo.xml /app/share/metainfo/com.github.neph_io.dsost.metainfo.xml
```
your app id is not com.github

bbhtt commented on com.github.neph_io.dsost.yml:
```diff
@@ -0,0 +1,39 @@
+    sources:
+      - type: git
+        url: https://github.com/Neph-IO/dsost.git
+        branch: main
```
branch not allowed, use tag or commit or both

bbhtt commented on com.github.neph_io.dsost.metainfo.xml:
```diff
@@ -0,0 +1,49 @@
+    </releases>
+    <content_rating type="oars-1.1" />
+
+    <developer id="com.github.neph_io">
```
com.github is not appid

bbhtt commented on com.github.neph_io.dsost.metainfo.xml:
```diff
@@ -0,0 +1,49 @@
+    <launchable type="desktop-id">com.github.neph_io.dsost.desktop</launchable>
+    <screenshots>
+        <screenshot type="default">
+            <image>https://raw.githubusercontent.com/Neph-IO/dsost/refs/heads/main/Screenshot.png</image>
```
this should be from a commit not main

bbhtt commented on com.github.neph_io.dsost.metainfo.xml:
```diff
@@ -0,0 +1,49 @@
+
+    <description>
+        <p>
+            Provide a simple Dark Souls OST streamer for you to enjoy the peacefull or agitated side of Lordran. Based on Python + GSstreamer.
```
this is barely a line, can you add some more description?

bbhtt commented on com.github.neph_io.dsost.desktop:
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Dark Souls OST
+Comment=Simple Dark Souls OST player
+Exec=python3 /app/src/main.py
```
this should be a proper launcher.

bbhtt commented:
why did you merge master here?

Neph-IO commented:
I have to be honest, i'm new with git and the merge was a mistake.

i renamed files and addressed all the changes according to the review comments.  (com.github was a mistake io.github was the way to go)
Thank you for your time !

bbhtt commented:
can you drop the merge commit?

Neph-IO commented:
My appologies i tried to revert my merge it look like it worked.

hfiguiere commented on io.github.neph_io.dsost.yml:
```diff
@@ -0,0 +1,37 @@
+  env:
+    - BASEAPP_REMOVE_WEBENGINE=1
+
+command: /app/bin/launch.sh
```
```suggestion
command: launch.sh
```

this should never be an absolute path.

hfiguiere commented on io.github.neph_io.dsost.yml:
```diff
@@ -0,0 +1,37 @@
+finish-args:
+  - --share=network
+  - --socket=pulseaudio
+  - --socket=x11
+  - --socket=wayland
```
either it supports wayland and it's `fallback-x11` or it is X11 only and then remove the wayland socket.

hfiguiere commented on io.github.neph_io.dsost.desktop:
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Dark Souls OST
+Comment=Simple Dark Souls OST player
+Exec=/app/bin/launch.sh
```
```suggestion
Exec=launch.sh
```

hfiguiere commented on io.github.neph_io.dsost.desktop:
this should be upstreamn

hfiguiere commented on io.github.neph_io.dsost.metainfo.xml:
this should be upstreamn

hfiguiere commented on io.github.neph_io.dsost.png:
this should be upstreamn
